src/DropAndBootstrapStakeFactory.sol

```diff
@@ -22,38 +23,20 @@ contract DropAndBootstrapStakeFactory {
 
         // no need to use salt as we want to keep the router address the same for the same fees &instructions
         bytes32 salt;
-
         DropAndBootstrapStakeRouter router;
-        try new DropAndBootstrapStakeRouter{ salt: salt }(fees, inst) returns (DropAndBootstrapStakeRouter router_) {
-            router = router_;
-        } catch {
-            router = DropAndBootstrapStakeRouter(
-                address(
-                    uint160(
-                        uint256(
-                            keccak256(
-                                abi.encodePacked(
-                                    bytes1(0xff),
-                                    address(this),
-                                    salt,
-                                    keccak256(
-                                        abi.encodePacked(
-                                            type(DropAndBootstrapStakeRouter).creationCode, abi.encode(fees, inst)
-                                        )
-                                    )
-                                )
-                            )
-                        )
-                    )
-                )
-            );
-        }
 
-        if (dropAmount > 0) {
-            inst.dropToken.safeTransferFrom(msg.sender, address(router), dropAmount);
+        bytes memory bytecode = abi.encodePacked(type(DropAndBootstrapStakeRouter).creationCode, abi.encode(fees, inst));
+        address routerAddr = Create2.computeAddress(salt, keccak256(bytecode));
+
+        if (routerAddr.code.length == 0) {
+            routerAddr = Create2.deploy(0, salt, bytecode);
+
+            if (dropAmount > 0) {
+                inst.dropToken.safeTransferFrom(msg.sender, routerAddr, dropAmount);
```

Comment:
if I put this line outside of `if (routerAddr.code.length == 0) {}`, when calling `deployDropAndBootstrapStakeRouter` for the second time (now the router already exists), it will still fail with `TransferFrom Failed`, any idea why is that? Why it won't topup the (existing) router and return?

Reply:
like this 
```solidity
        bytes memory bytecode = abi.encodePacked(type(DropAndBootstrapStakeRouter).creationCode, abi.encode(fees, inst));
        address routerAddr = Create2.computeAddress(salt, keccak256(bytecode));

        if (routerAddr.code.length == 0) {
            routerAddr = Create2.deploy(0, salt, bytecode);
        }

        if (dropAmount > 0) {
            // when contract exist, transferFrom Failed!
            inst.dropToken.safeTransferFrom(msg.sender, routerAddr, dropAmount);
        }

        return DropAndBootstrapStakeRouter(routerAddr);
```